pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/sink/PulsarSink.java

```diff
@@ -150,11 +148,11 @@ public Producer<T> createProducer(PulsarClient client, String topic, String prod
         }
 
         protected Producer<T> getProducer(String destinationTopic, Schema schema) {
-            return getProducer(destinationTopic, null, destinationTopic, schema);
+            return getProducer(destinationTopic, schema, null, null);
         }
 
-        protected Producer<T> getProducer(String producerId, String producerName, String topicName, Schema schema) {
-            return publishProducers.computeIfAbsent(producerId, s -> {
+        protected Producer<T> getProducer(String topicName, Schema schema, String producerName, String partitionId) {
```

Comment:
I removed the `producerId` since there isn't one in the refactored solution. Having the topic name as the first parameter is meaningful after the refactoring.